### PR TITLE
get the go version from the .go-version file

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

The partial migration of the Jenkins pipeline to the Buildkite pipeline. 

## How does this PR solve the problem?

Fix of the getting go-version

## How to test this PR locally

It doesn't be checked locally, that's why it's a partial "step by step" migration

## Checklist

The Buildkite pipeline will be triggered by this PR and we will see the result

## Related issues

Related to the issue: https://github.com/elastic/fleet-server/issues/2517

### CODE REVIEW

File: .buildkite/pull-requests.json [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-        "set_commit_status": false,
-        "build_on_commit": false,
-        "build_on_comment": true,
-        "trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:package)\\W+(?:this|it))|^/package$",
-        "always_trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:package)\\W+(?:this|it))|^/package$",
-
-        "fail_on_not_mergeable": true
-      }]
-}
+{
```
Comment (mrodm) [inline .buildkite/pull-requests.json:1]: what is the change in this file ?

File: .buildkite/hooks/post-checkout [hunk 2]
```diff
@@ -50,5 +50,8 @@ echo "Commit information"
 
 git --no-pager log --format=%B -n 1
 
+export GO_VERSION="$(< .go-version)"
+echo "$GO_VERSION"
+
```
Comment (mrodm) [inline .buildkite/hooks/post-checkout:55]: echoing the go version is just for debugging ? could it be removed?

**Review by mrodm — COMMENTED** [review]
If defining environment varaibles can be used in the step definitoin, maybe it could be then updated also in other repositories. And just trust `.go-version`.

There was a related PR to add the needed configuration in the CLI config to update the Golang version from the pipelines

https://github.com/elastic/fleet-server/pull/2714

If this works, that config entry could be removed too

File: .buildkite/pull-requests.json (hunk 1 above)
Comment (sharbuz) [reply .buildkite/pull-requests.json:1]: In this file was changed just EOL from CRLF to LF

**Pull request merged**